data_structures/segment_tree/segment_tree.c

```diff
@@ -0,0 +1,76 @@
+#include <string.h>
+#include "segment_tree.h"
+
+/**
+ * Structure of segment tree
+*/
+struct segment_tree
+{
+    comp_func_t comp_func;      // compare function
```

Comment:
```suggestion
    comp_func_t comp_func;      /**< compare function */
```
In doxygen standard, the structure members are documented like thus.